aspnetcore/blazor/forms-validation.md

```diff
@@ -223,12 +230,383 @@ In the following example:
+        private EditContext CurrentEditContext { get; set; }
+
+        protected override void OnInitialized()
+        {
+            if (CurrentEditContext == null)
+            {
+                throw new InvalidOperationException(
+                    $"{nameof(ServerValidator)} requires a cascading " +
+                    $"parameter of type {nameof(EditContext)}. " +
+                    $"For example, you can use {nameof(ServerValidator)} " +
+                    $"inside an {nameof(EditForm)}.");
+            }
+
+            messageStore = new ValidationMessageStore(CurrentEditContext);
+
+            CurrentEditContext.OnValidationRequested += (s, e) => 
+                messageStore.Clear();
+            CurrentEditContext.OnFieldChanged += (s, e) => 
+                messageStore.Clear(e.FieldIdentifier);
+        }
+
+        public void DisplayErrors(Dictionary<string, List<string>> errors)
+        {
+            foreach (var err in errors)
+            {
+                messageStore.Add(CurrentEditContext.Field(err.Key), err.Value);
+            }
 
-        return serverChecksValid;
+            CurrentEditContext.NotifyValidationStateChanged();
+        }
+    }
+}
+```
+
+In the client project, the *Starfleet Starship Database* form is updated to show server validation errors with the `ServerValidaor` component. When the server API returns validation messages, they're added to the form's <xref:Microsoft.AspNetCore.Components.Forms.ValidationMessageStore>:
+
+```csharp
+@page "/FormsValidation"
+@using System.Net.Http.Json
+@using Microsoft.AspNetCore.Authorization
+@using Microsoft.AspNetCore.Components.WebAssembly.Authentication
+@using {SOLUTION NAMESPACE}.Shared
+@attribute [Authorize]
+@inject HttpClient Http
+
+<h1>Starfleet Starship Database</h1>
+
+<h2>New Ship Entry Form</h2>
+
+<EditForm Model="@starship" OnValidSubmit="HandleValidSubmit">
+    <DataAnnotationsValidator />
+    <ServerValidator @ref="serverValidator" />
+    <ValidationSummary />
+
+    ...
+
+</EditForm>
+
+@code {
+    private ServerValidator serverValidator;
+    private Starship starship = new Starship() { ProductionDate = DateTime.UtcNow };
+
+    private async Task HandleValidSubmit(EditContext editContext)
+    {
+        try
+        {
+            var response = await Http.PostAsJsonAsync<Starship>(
+                "StarshipValidation", (Starship)editContext.Model);
+
+            var errors = await response.Content
+                .ReadFromJsonAsync<Dictionary<string, List<string>>>();
+
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new HttpRequestException("Server validation failed");
+            }
+            else if (errors.Count() > 0)
+            {
+                serverValidator.DisplayErrors(errors);
+            }
+            else
+            {
+                // Process the form
+            }
+        }
+        catch (AccessTokenNotAvailableException ex)
```

Comment:
This is noise in this documentation

Reply:
I feel like this ... and the exception trap ... are virtually required by what we know that devs do ... they **_cut-'n-paste code directly out of our examples into their projects_** 😨😄 and never give it a second thought. Given our other guidance, it's safer for us to leave this here. It matches what we'd like them to do when, for example, they're following our Azure guidance for a hosted Blazor solution.